### PR TITLE
[Qt5 C++] Replace illegal characters in param names for Qt5 generator

### PR DESCRIPTION
**Pull request by Ragnis**
### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Replace any invalid characters in param names.

### CODE REVIEW

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/Qt5CPPGenerator.java [hunk 1]
```diff
@@ -337,6 +337,11 @@ public String toModelName(String type) {
     }
 
     @Override
+    public String toParamName(String name) {
+        return name.replaceAll("[^A-Za-z0-9_]", "_").replaceAll("_+", "_");
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/Qt5CPPGenerator.java]: One suggestion: What about reusing similar logic in Ruby toParamName/toVarName to remove illegal characters and convert the name into snake_case? https://github.com/swagger-api/swagger-codegen/blob/1dd9ee39af6c4e2f9c016a94d16ca47a88a9cb0d/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/RubyClientCodegen.java#L434-L459

**Comment by Ragnis** [comment]
I have updated it to use the sanitization logic from Ruby code.

**Comment by wing328** [comment]
@Ragnis OK. I'll take a look later and merged accordingly.

**Comment by wing328** [comment]
Looks good to me. I noticed that you also fixed the {{paramName}} with {{baseName}} 👍

**Comment by wing328** [comment]
@Ragnis PR merged into master. Thanks for the contribution.

**Pull request merged**